src/index.js

```diff
@@ -169,7 +172,7 @@ export default function renderToString(vnode, context, opts, inner) {
 		for (let i=0; i<len; i++) {
 			let child = children[i];
 			if (!falsey(child)) {
-				let ret = renderToString(child, context, opts, true);
+				let ret = renderToString(child, context, opts, true, nodeName==='svg');
```

Comment:
as for the test - I'd do something like that but take it a little further and verify that it's active/inactive for descendants:

```js
// input:
<svg>
  <image xlinkHref="#foo" />
  <foreignObject>
    <div contentEditable />
  </foreignObject>
</svg>

// output:
`<svg>
  <image xlink:href="#foo" />
  <foreignObject>
    <div contentEditable />
  </foreignObject>
</svg>`
```